### PR TITLE
Problem: boilerplate when init msg from data copy

### PR DESCRIPTION
**Pull request by gummif**
Solution: Add zmq_msg_init_buffer to construct
a message by copying memory from buffer.

See issue https://github.com/zeromq/libzmq/issues/3858

### CODE REVIEW

**Comment by bluca** [comment]
THis is a new API, so it must go in the DRAFT section of the header and in src/zmq_drafts.h. Also it's probably better to avoid renaming existing tests - people might have them on SKIP individually

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -18,6 +18,7 @@ set(tests
   test_invalid_rep
   test_msg_flags
   test_msg_ffn
+  test_msg_init
```
Comment (bluca) [inline tests/CMakeLists.txt:21]: needs to be in the DRAFT section

File: Makefile.am [hunk 2]
```diff
@@ -451,6 +451,7 @@ test_apps = \
 	tests/test_invalid_rep \
 	tests/test_msg_flags \
 	tests/test_msg_ffn \
+	tests/test_msg_init \
```
Comment (bluca) [inline Makefile.am:454]: needs to be in the draft section or it won't build

File: tests/CMakeLists.txt (hunk 1 above)
Comment (gummif) [reply tests/CMakeLists.txt]: fixed

**Pull request merged**